### DISCUSSION TITLE
Are there example of tuning syntax regarding new tune detection feature?

### DISCUSSION THREAD

**Discussion by Syngelik** [Security-Onion-Solutions/securityonion#13198] [General] [answered]
As the title indicates, I'm looking for examples when tuning, specifically when using the modify choice. In my case, I have a couple rules where I'd like a couple parent domains to not trigger perl or curl user-agent rules. If I wanted to "whitelist" say microsoft.com, where can I find what that would look like?

**Comment by cm-ops** [comment]
You could add a content keyword https://docs.suricata.io/en/suricata-7.0.5/rules/payload-keywords.html something like ` content: !"microsoft.com";`.

**Reply by Syngelik** [reply]
Right, am I able to do that in the web console or do I need to edit the all.rules file on the backend?

**Reply by cm-ops** [reply]
You would do that in SOC, when you select tune detection and modify, there is a line for the regex and line for the value you are modifying/adding.

**Reply by Syngelik** [reply]
Circling back around on this, any chance you can give an explicit example of what would go in each field?

I'm clearly doing something silly.

**Reply by sutehk-cs** [reply]
I would appreciate this as well. I've tried all kinds of inputs to no avail. I've defaulted to editing the rule itself for now.

**Reply by cm-ops** [reply]
@Syngelik give me an example SID please.

**Reply by Syngelik** [reply]
@cm-ops The original rules that I was trying to modify with the tune detection tool were 2013028, 2013030, or 2013031

**Reply by cm-ops** [reply, marked as answer]
![image](https://github.com/user-attachments/assets/86560f87-7ee8-40cb-a420-ed1b6a1ae8b9)

Rule after modify:
```
[root@so-standalone ~]# grep 2013028 /opt/so/rules/nids/suri/all.rules
alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"ET POLICY curl User-Agent Outbound"; flow:established,to_server; http.user_agent; content:"curl/"; nocase; startswith; content: !"microsoft.com"; nocase; reference:url,www.useragentstring.com/pages/useragentstring.php; classtype:attempted-recon; sid:2013028; rev:8; metadata:created_at 2011_06_14, deprecation_reason Performance, updated_at 2024_07_02;)
```
